shared-module/bitmaptools/__init__.c

```diff
@@ -918,3 +921,59 @@ void common_hal_bitmaptools_alphablend(displayio_bitmap_t *dest, displayio_bitma
         }
     }
 }
+
+STATIC void draw_circle(displayio_bitmap_t *destination,
+    int16_t x0, int16_t y0,
+    int16_t radius, uint32_t value) {
+
+    int16_t d, y;
+
+    mp_arg_validate_int_range(x0, SHRT_MIN, SHRT_MAX, MP_QSTR_x0);
```

Comment:
Should these be validated against the size of the `bitmap` or do  you want them to be able to draw it "off" screen (where I realize some may show up on the current screen.

Reply:
Hello :). First I want to say, that my comments are intended to seek advice/guidance. The core, is not the place I spent most of my time ;). 
Intention: Draw of the scree, I was just following the logic in vectorio, I have no preference on which path to go. We could add the check equivalent to this
```
        x0 = MAX(0, x0); // only draw inside bitmap
        x1 = MIN(x1, destination->width - 1);

```

Reply:
Drawing only the portion that ends up visible on the screen based on the size and x/y location that the user passes in is good I think. 

Currently behavior drawing on the edges looks good to me.

Reply:
Yes, I will change the behaviour to verify with width->height and displayu->height

Reply:
@gamblor21 and @FoamyGuy after watching the stream, I will do some testing. As you probably know, The intention of this draw_circle is to have a shell. Vectorio, does have a circle but is by default `filled`. I am going to test if using bitmaptools.fill in this circle will throw an exception is the circle is not complete. aka we limit the circle to the display option.  Also. I will some example code as draw_polygon. The problem with the constantly rebooting was do to a previous PR now fixed. See discord dev channel for more info. I spent some time trying to find the root cause as well as @dhalbert :) Thanks y'all for the reviews